translate/convert/l20n2po.py

```diff
@@ -0,0 +1,117 @@
+# along with this program; if not, see <http://www.gnu.org/licenses/>.
+
+
+"""Convert Mozilla .l20n files to Gettext PO localization files.
+
+See: http://docs.translatehouse.org/projects/translate-toolkit/en/latest/commands/l20n2po.html
+for examples and usage instructions.
+"""
+
+from translate.storage import po, l20n
+
+
+class l20n2po(object):
+    """convert a .l20n file to a .po file for handling the
+    translation."""
+
+    def __init__(self, blankmsgstr=False, duplicatestyle="msgctxt"):
+        self.blankmsgstr = blankmsgstr
+        self.duplicatestyle = duplicatestyle
+
+    def convert_l20nunit(self, unit):
+        po_unit = po.pounit(encoding="UTF-8")
+        po_unit.setid(unit.getid())
+        po_unit.addlocation(unit.getid())
+        po_unit.source = unit.value
+        po_unit.addnote(unit.comment, "developer")
+
+        return po_unit
+
+    def convert_store(self, l20n_store):
+        """converts a .l20n file to a .po file..."""
+        target_store = po.pofile()
+        targetheader = target_store.header()
+        targetheader.addnote("extracted from %s" % l20n_store.filename,
+                             "developer")
+        l20n_store.makeindex()
+        for l20nunit in l20n_store.units:
+            pounit = self.convert_l20nunit(l20nunit)
+            target_store.addunit(pounit)
+        target_store.removeduplicates(self.duplicatestyle)
+        return target_store
+
+    def merge_stores(self, origl20nfile, translatedl20nfile):
+        """converts two .l20n files to a .po file..."""
+        target_store = po.pofile()
+        targetheader = target_store.header()
+        targetheader.addnote(
+            "extracted from %s, %s" % (origl20nfile.filename,
+                                       translatedl20nfile.filename),
+            "developer"
+        )
+        translatedl20nfile.makeindex()
+        for l20nunit in origl20nfile.units:
+            pounit = self.convert_l20nunit(l20nunit)
+            target_store.addunit(pounit)
+            for location in pounit.getlocations():
+                # TODO: check if using of `id_index` fits here in all cases
+                if location in translatedl20nfile.id_index:
+                    l20nunit = translatedl20nfile.id_index[location]
+                    pounit.target = l20nunit.target
+        target_store.removeduplicates(self.duplicatestyle)
+        return target_store
+
+
+def convertl20n(inputfile, outputfile, templatefile,
+                pot=False, duplicatestyle="msgctxt"):
+    inputstore = l20n.l20nfile(inputfile)
+    convertor = l20n2po(blankmsgstr=pot, duplicatestyle=duplicatestyle)
+    if templatefile is None:
+        outputstore = convertor.convert_store(inputstore)
+    else:
+        templatestore = l20n.l20nfile(templatefile)
+        outputstore = convertor.merge_stores(templatestore, inputstore)
+    if outputstore.isempty():
+        return 0
+    outputstore.serialize(outputfile)
+    return 1
```

Comment:
@unho there is a mess in convertors. can we just land l20n stuff and address cleanup concerns later?